packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -0,0 +1,33 @@
+/**
+ * A base class for `Error`s thrown by GOV.UK Frontend.
+ *
+ * It is meant to be extended into specific types of errors
+ * to be thrown by our code.
+ *
+ * @example
+ * ```js
+ * class MissingRootError extends GOVUKFrontendError {
+ *   // Setting an explicit name is important as extending the class will not
+ *   // set a new `name` on the subclass. The `name` property is important
+ *   // to ensure intelligible error names even if the class name gets
+ *   // mangled by a minifier
+ *   name = "MissingRootError"
+ * }
+ * ```
+ * @abstract
+ */
+export class GOVUKFrontendError extends Error {
+  name = 'GOVUKFrontendError'
```

Comment:
Annoying, but necessary. Simply extending `Error` doesn't give the error a new name, unfortunately. 

And going for something automated like `this.name = this.constructor.name` in `GOVUKFrontendError`'s `constructor` means the error gets a mangled name once minifier have run over the code and rename the class to say `n`. While we may prevent the mangling in our files, we don't control how services minify theirs, so hardcoding the name limits risks of having an error with the name of `n`.